src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -395,11 +403,12 @@ private void emitGlobalGetter(String holderClass, String javaName,
         incrAlign();
         emitDocComment(decl, docHeader);
         Class<?> type = Utils.carrierFor(decl.type());
-        appendLines(STR."""
-            public static \{type.getSimpleName()} \{javaName}() {
-                return \{holderClass}.SEGMENT.get(\{holderClass}.LAYOUT, 0L);
+        appendLines(String.format("""
+            public static %s %s() {
+                return %s.SEGMENT.get(%s.LAYOUT, 0L);
             }
-            """);
+            """, type.getSimpleName(), javaName, holderClass, holderClass));
```

Comment:
This might also help improving readability for messages containing lots of `%s` (as at least you can look at the number being referenced, and then count the arguments passed to the `format` method to find the one you want. Also, this allows to refactor code more robustly - e.g. you can pass _new_ trailing parameters to `format`, and maybe have them used _before_ the others:

```
jshell> MessageFormat.format("{0} {2} {0} {1}", "Hello", "World!", "Maurizio");
$7 ==> "Hello Maurizio Hello World!"
```

E.g. thanks to explicit numbers, the formatting is not tied to the order in which the arguments are passed to the `format` method.